ui/app/components/app/signature-request-original/signature-request-original.component.js

```diff
@@ -0,0 +1,325 @@
+    let notice = this.context.t('youSign') + ':'
+
+    const { txData } = this.props
+    const { type, msgParams: { data } } = txData
+
+    if (type === 'personal_sign') {
+      rows = [{ name: this.context.t('message'), value: this.msgHexToText(data) }]
+    } else if (type === 'eth_signTypedData') {
+      rows = data
+    } else if (type === 'eth_sign') {
+      rows = [{ name: this.context.t('message'), value: data }]
+      notice = (
+        <React.Fragment>
+          { this.context.t('signNotice') }
+          <span
+            className="request-signature__help-link"
+            onClick={() => {
+              global.platform.openWindow({
+                url: 'https://metamask.zendesk.com/hc/en-us/articles/360015488751',
+              })
+            }}
+          >
+            { this.context.t('learnMore') }
+          </span>
+        </React.Fragment>
+      )
+    }
+
+    return (
+      <div className="request-signature__body">
+        { this.renderAccountInfo() }
+        { this.renderRequestInfo() }
+        <div
+          className={classnames({
+            'request-signature__notice': type === 'personal_sign' || type === 'eth_signTypedData',
+            'request-signature__warning': type === 'eth_sign',
+          })}
+        >
+          { notice }
+        </div>
+        <div className="request-signature__rows">
+          {
+            rows.map(({ name, value }, index) => {
+              if (typeof value === 'boolean') {
+                value = value.toString()
+              }
+              return (
+                <div className="request-signature__row" key={`request-signature-row-${index}`}>
+                  <div className="request-signature__row-title">
+                    { `${name}:` }
+                  </div>
+                  <div className="request-signature__row-value">
+                    { value }
+                  </div>
+                </div>
+              )
+            })
+          }
+        </div>
+      </div>
+    )
+  }
+
+  renderFooter = () => {
+    const { cancel, sign } = this.props
+
+    return (
+      <div className="request-signature__footer">
+        <Button
+          type="default"
+          large
+          className="request-signature__footer__cancel-button"
+          onClick={event => {
+            this._removeBeforeUnload()
+            cancel(event).then(() => {
+              this.context.metricsEvent({
+                eventOpts: {
+                  category: 'Transactions',
+                  action: 'Sign Request',
+                  name: 'Cancel',
+                },
+              })
+              this.props.clearConfirmTransaction()
+              this.props.history.push(DEFAULT_ROUTE)
+            })
+          }}
+        >
+          { this.context.t('cancel') }
+        </Button>,
+        <Button
+          type="secondary"
+          large
+          className="request-signature__footer__sign-button"
+          onClick={event => {
```

Comment:
Can we make this handler `async` and `await` the completion of `sign(event) below?